src/ray/core_worker/store_provider/plasma_store_provider.cc

```diff
@@ -319,6 +319,12 @@ Status CoreWorkerPlasmaStoreProvider::Get(
         FetchAndGetFromPlasmaStore(remaining, batch_ids, batch_timeout,
```

Comment:
alternatively, we can change the behavior of FetchAndGetFromPlasmaStore behavior on raylet. let me know which makes more sense.

Reply:
Changing FetchAndGet sounds better to me

> cc @ericl IIUC, this might not go well with our fallback allocation now (meaning we should choose the alternative approach)? (Since now we distinguish ray.get & ray.wait requests)? Can you confirm?

Yeah, we shouldn't use Wait() since that won't have as high priority as get in PullManager, and also we treat individual object ids in Wait separately instead of trying to as a bundle in PullManager.

Reply:
Can you remind me where the infinite loop is?